basic/src/test/java/org/jboss/hal/testsuite/test/configuration/elytron/ElytronFactoryTestCase.java

```diff
@@ -0,0 +1,1550 @@
+            new ResourceVerifier(saslServerFactoryAddress1, client).verifyExists();
+            new ResourceVerifier(saslServerFactoryAddress2, client).verifyExists();
+
+            WizardWindow wizard = page.getResourceManager().addResource();
+            Editor editor = wizard.getEditor();
+            editor.text(NAME, aggregateFactoryName);
+            editor.text(SASL_SERVER_FACTORIES, saslServerFactoryNamesSeparatedByNewLine);
+            boolean closed = wizard.finish();
+
+            assertTrue("Dialog should be closed!", closed);
+            assertTrue("Created Aggregate SASL Server Mechanism should be present in the table!",
+                    page.resourceIsPresentInMainTable(aggregateFactoryName));
+
+            ModelNode expectedHttpFactoriesNode = new ModelNodeListBuilder()
+                    .addNode(new ModelNode(saslServerFactoryName1)).addNode(new ModelNode(saslServerFactoryName2))
+                    .build();
+            new ResourceVerifier(aggregateFactoryAddress, client).verifyExists().verifyAttribute(SASL_SERVER_FACTORIES,
+                    expectedHttpFactoriesNode);
+        } finally {
+            ops.removeIfExists(aggregateFactoryAddress);
+            ops.removeIfExists(saslServerFactoryAddress1);
+            ops.removeIfExists(saslServerFactoryAddress2);
+            adminOps.reloadIfRequired();
+        }
+    }
+
+    @Test
+    public void editAggregateSaslServerTest() throws Exception {
+        final String saslServerFactoryName1 = RandomStringUtils.randomAlphanumeric(5),
+                saslServerFactoryName2 = RandomStringUtils.randomAlphanumeric(5),
+                saslServerFactoryName3 = RandomStringUtils.randomAlphanumeric(5),
+                saslServerFactoryName4 = RandomStringUtils.randomAlphanumeric(5),
+                newHttpServerFactoryNamesSeparatedByNewLine = saslServerFactoryName3 + "\n" + saslServerFactoryName4,
+                aggregateFactoryName = RandomStringUtils.randomAlphanumeric(5);
+        final Address saslServerFactoryAddress1 = elyOps.getElytronAddress(SERVICE_LOADER_SASL_SERVER_FACTORY,
+                saslServerFactoryName1),
+                saslServerFactoryAddress2 = elyOps.getElytronAddress(SERVICE_LOADER_SASL_SERVER_FACTORY,
+                        saslServerFactoryName2),
+                saslServerFactoryAddress3 = elyOps.getElytronAddress(PROVIDER_SASL_SERVER_FACTORY,
+                        saslServerFactoryName3),
+                        saslServerFactoryAddress4 = elyOps.getElytronAddress(PROVIDER_SASL_SERVER_FACTORY,
+                                saslServerFactoryName4),
+                aggregateFactoryAddress = elyOps.getElytronAddress(AGGREGATE_SASL_SERVER_FACTORY,
+                        aggregateFactoryName);
+        final ModelNode initialHttpFactoriesNode = new ModelNodeListBuilder()
+                .addNode(new ModelNode(saslServerFactoryName1)).addNode(new 
```

Comment:
Why `Assert#fail()` and not `@Ignore`? If this is OK, I would welcome if this `TODO` would be accompanied by explanation.